pkg/helpers/newapp/app/sourcelookup.go

```diff
@@ -76,16 +76,11 @@ func IsRemoteRepository(s string) (bool, error) {
 		klog.V(5).Infof("%s is not a valid remote git clone spec", s)
 		return false, nil
 	}
-	gitRepo := git.NewRepository()
+	// don't need to check error since we already call s2igit.Parse
+	sri, _ := NewSourceRepository(s, newapp.StrategyUnspecified)
+	err = sri.DetectAuth()
```

Comment:
I think we need a comment here explaining what `DetectAuth` does, and why that helps us determine if we have a valid remote git repository.

Reply:
sure will add